lib/src/internal/FlowManager.cpp

```diff
@@ -226,23 +278,63 @@ namespace mxl::lib
         // Verify that the flow file exists.
         if (auto const flowFile = makeFlowDataFilePath(base); exists(flowFile))
         {
-            auto flowSegment = SharedMemoryInstance<Flow>{flowFile.string().c_str(), in_mode, 0U};
-            if (auto const flowFormat = flowSegment.get()->info.common.format; mxlIsDiscreteDataFormat(flowFormat))
+            auto flowSegment = SharedMemoryInstance<Flow>{};
+            try
             {
-                return openDiscreteFlow(base, std::move(flowSegment));
+                flowSegment = SharedMemoryInstance<Flow>(flowFile.string().c_str(), in_mode, 0U);
+            }
+            catch (std::exception const& e)
+            {
+                MXL_ERROR("FlowManager: Failed to open flow data segment '{}': {}", flowFile.string(), e.what());
+                throw std::filesystem::filesystem_error(
+                    "FlowManager: Failed to open flow data segment", flowFile, std::make_error_code(std::errc::io_error));
+            }
```

Comment:
1. As above wrt to rethrowing the original exception.
2. I tend to prefer the conciseness of the original and am not sure whether it's worth sacrificing this for the added log statement alone.